kbase-extension/static/kbase/js/common/jobs.js

```diff
@@ -1,37 +1,129 @@
-define([
-    'common/html',
-    'common/format',
-    'common/ui',
-], (
-    html,
-    format,
-    UI,
-) => {
+define(['common/html', 'common/format', 'common/ui'], (html, format, UI) => {
     'use strict';
 
     const t = html.tag,
-        span = t('span'),
-        cssBaseClass = 'kb-job-state',
+        span = t('span');
+
+    const cssBaseClass = 'kb-job-state',
         jobNotFound = [
             'This job was not found, or may not have been registered with this Narrative.',
-            'You will not be able to inspect the job status or view the job log'
+            'You will not be able to inspect the job status or view the job log',
         ],
-        jobStatusUnknown = ['Determining job state...'];
+        jobStatusUnknown = ['Determining job state...'],
+        // valid job states from ee2, plus 'does_not_exist'
+        validJobStates = [
+            'created',
+            'estimating',
+            'queued',
+            'running',
+            'completed',
+            'error',
+            'terminated',
+            'does_not_exist',
+        ];
+
+    /**
+     * A jobState object is deemed valid if
+     * 1. It's an object (not an array or atomic type)
+     * 2. It has a 'created' key
+     * 3. It has a 'job_id' key
+     * 4. It has a 'status' key, which appears in the array validJobStates
```

Comment:
tighten up the definition of a valid job by ensuring that the state is one of those listed